### PR TITLE
Add release notes for 1.77.0

### PR DESCRIPTION
**Pull request by cuviper**
cc @rust-lang/release 
r? @Mark-Simulacrum

### CODE REVIEW

**Comment by cuviper** [comment]
Still some TODOs for a few T-types changes I haven't figured out yet, and choosing compat notes (if any).

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,127 @@
...
+- [fix fn/const items implied bounds and wf check (rebase)](https://github.com/rust-lang/rust/pull/120019/)
+
+<a id="1.77.0-Compiler"></a>
+
+Compiler
+--------
+
+- [Include lint `soft_unstable` in future breakage reports.](https://github.com/rust-lang/rust/pull/116274/)
+- [Make `i128` and `u128` 16-byte aligned on x86-based targets.](https://github.com/rust-lang/rust/pull/116672/)
+- [Add lint `static_mut_ref` to warn on references to mutable statics.](https://github.com/rust-lang/rust/pull/117556/)
```
Comment (Urgau) [inline RELEASES.md:28]: The [linked issue](https://github.com/rust-lang/rust/issues/114447) in the PR has `T-lang` labels. Should it be moved to the language section?

The lint was also renamed to `static_mut_refs` in https://github.com/rust-lang/rust/pull/121034.

```suggestion
- [Add lint `static_mut_refs` to warn on references to mutable statics.](https://github.com/rust-lang/rust/pull/117556/)
```

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,127 @@
...
+- [Undeprecate lint `unstable_features` and make use of it in the compiler.](https://github.com/rust-lang/rust/pull/118639/)
```
Comment (Urgau) [inline RELEASES.md:29]: This also feels more language to me. The [FCP was done](https://github.com/rust-lang/rust/pull/118639/#issuecomment-1864781050) by T-lang.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,127 @@
...
+<a id="1.77.0-Language"></a>
+
+Language
+--------
+
+- [Reveal opaque types within the defining body for exhaustiveness checking.](https://github.com/rust-lang/rust/pull/116821/)
+- [Stabilize C-string literals.](https://github.com/rust-lang/rust/pull/117472/)
+- [Stabilize THIR unsafeck.](https://github.com/rust-lang/rust/pull/117673/)
+- [Support async recursive calls (as long as they have indirection).](https://github.com/rust-lang/rust/pull/117703/)
+- [Get rid of type-driven traversal in const-eval interning.](https://github.com/rust-lang/rust/pull/119044/)
```
Comment (cuviper) [inline RELEASES.md:13]: Note to self: T-lang is reconsidering this one: https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/should.20we.20revert.20the.20removal.20of.20type-driven.20traversal.20.2312161

**Comment by cuviper** [comment]
Should be ready for final review now.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,120 @@
...
+- [`RwLock::clear_poison`](https://doc.rust-lang.org/stable/std/sync/struct.RwLock.html#method.clear_poison)
+
+<a id="1.77.0-Cargo"></a>
+
+Cargo
+-----
+
+- [Extend the build directive syntax with `cargo::`.](https://github.com/rust-lang/cargo/pull/12201/)
+- [Stabilize metadata `id` format as `PackageIDSpec`.](https://github.com/rust-lang/cargo/pull/12914/)
+- [Pull out as `cargo-util-schemas` as a crate.](https://github.com/rust-lang/cargo/pull/13178/)
```
Comment (mati865) [inline RELEASES.md:85]: Sorry for the late comment.
This was probably intended without the first `as`:
```suggestion
- [Pull out `cargo-util-schemas` as a crate.](https://github.com/rust-lang/cargo/pull/13178/)
```

Comment (cuviper) [reply RELEASES.md:85]: Oops! See #122699, but I think we don't need to bother respinning the stable release.

Comment (mati865) [reply RELEASES.md:85]: Blogs link to https://doc.rust-lang.org/nightly/releases.html anyway so indeed it might be not worth the trouble.

**Pull request merged**